sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -293,6 +313,39 @@ should take ownership of the passed native handle. The default behavior is to
 transfer the ownership to the SYCL runtime. See section 4.4 for details. If
 the behavior is "transfer" then the runtime is going to destroy the input
 Level-Zero kernel</td>
+</tr><tr>
+<td>
+
+``` C++
+make_buffer(
+    const backend_input_t<backend::ext_oneapi_level_zero,
+                          buffer<T, Dimensions, AllocatorT>> &,
+    const context &Context)
+```
+</td>
+<td>This API is available starting with revision 2 of this specification.
+
+Construct a SYCL buffer instance from a pointer to a Level Zero memory buffer. The pointer must be the value returned from a previous call to <code>zeMemAllocShared()</code>, <code>zeMemAllocDevice()</code>, or <code>zeMemAllocHost()</code>. If pointer is the value returned from <code>zeMemAllocDevice()</code> or <code>zeMemAllocShared()</code> then SYCL context <code>Context</code> must be associated with a single device.
+The <code>Context</code> argument must be a valid SYCL context encapsulating a Level-Zero context, and the Level-Zero memory must be allocated on the same context. Created SYCL buffer is
+associated only with the single SYCL context <code>Context</code>, the buffer can't be accessed on another contexts.

```

Comment:
Actually, I've realized that we don't need this limitation at all. 

Initially I made this restriction because memory buffer can be used only in the context where it is created: 
https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=zememallocdevice#zememallocdevice

> The application must only use the memory allocation for the context and device, or its sub-devices, which was provided during allocation.

But it turns out SYCL RT handles the case when SYCL buffer is used in different context (opencl buffer is also isolated in the context).  So, if buffer encapsulates allocation on ctx1 and this buffer is accessed on ctx2 then SYCL RT creates a new allocation on ctx2 and copies contents of the first allocation to the second allocation through host (i.e. 2 copy operations - device to host and host to device)

But there is one problem. After buffer destruction data is not automatically copied back to original allocation on ctx1 (to follow keep semantics that you described). Currently contents of the buffer can be accessed through host accessor (or probably somehow artificially by submitting buffer to the original context with empty kernel).

Copy back logic requires additional changes in buffer and/or scheduler. I am not sure if we can somehow describe this as a limitation and implement later.

Reply:
I think we should avoid documenting a feature that we don't implement.  This will just lead to customer confusion and bug reports.  Here are a couple of options:

1. Add the support now to copy back the data to the original buffer for this case.

2. Support only the `transfer` case in this version of the extension.  This is the default behavior anyway, so presumably we think this is the common case.  We could retain the `Ownership` field, but `make_buffer` would throw an exception if the user passes `keep`.  Since the issue about copying data back only exists in the "keep" case, we don't need to implement that now.  We have the option in the future to add support for `keep` and implement the copy-back behavior in the future.

Reply:
But "keep" is only "broken" on case of access to the buffer from multiple contexts, which would be a very rare case, I believe. I'd vote for documenting the "keep" and make it work while in the same context, and create a bug about missing copy for multiple contexts.

Reply:
If we think the "keep" case is important, we should implement it for all cases.  How hard is it to do the write-back?

Reply:
Implemented write-back and added test to https://github.com/intel/llvm-test-suite/pull/708